lib/chef/knife/kvm_base.rb

```diff
@@ -33,6 +33,7 @@ def run_remote_command(command)
               if success
                 channel.on_data do |ch, data|
                   return_val = data.chomp
+                  print return_val if console
```

Comment:
if I recall ui.* methods end up doing a newline and don't print every character? I will confess I didn't spend much time debugging with them. If I recall correctly - it did something weird so I went lowest common denominator. I can give it a shot with ui stuff.